### ISSUE TITLE
Investigate authsome library

### ISSUE THREAD

**Issue by Cooryd** [elifesciences/elife-xpub#1993]
## Task

As none of us are hugely familiar with authsome. It would be useful to do a deep dive into the authsome repo to get to know the library before we begin adding it to the app. 

https://gitlab.coko.foundation/pubsweet/authsome

_Give Jure a prod if anything doesn't make sense_

**Comment by diversemix** [comment]
Looking at the current implementation of [`withAuthsomeMiddleware.js`](https://github.com/elifesciences/elife-xpub/blob/develop/packages/component-elife-app/server/withAuthsomeMiddleware.js) ... there are a number of things that are apparent:

1. Permissions are defined in code and not in configuration / data
2.  Permissions are for a user rather than a group - this could be overcome by the permissions querying the groups for the users - but it does muddy the permissions model by it needing to know about the User/Team model.
3. Permissions are tied to the model. To clarify this is the data model, not the domain model. Again the permission logic could also use the domain model - but this is again muddying the permission logic with domain knowledge.
4. The verb are nominally linked to the graphQL actions. Again a good first guess but access to the API (however it is implemented) is an infrastructure issue. What we require is an expression of domain level permissions. 

In conclusion we want to be able to express our permissions in terms of domain objects, verbs etc and so I would suggest we define these in the domain model and store them in the database. I will raise an ADR for a suggestion on how this is done.

/cc @giorgiosironi @Cooryd @hdrury1 @jure @will-byrne @erezmus @petereast

**Comment by diversemix** [comment]
I'm going to close this and continue the discussion on #1987
